### PR TITLE
Feat: Reducer functions

### PR DESCRIPTION
**Pull request by yashpatel1797**
Added Reducer functions
1. Authentication
2. Cart management
3. Wishlist management
4. Filter management

[Live link](https://joyful-rolypoly-31b73d.netlify.app/)

### CODE REVIEW

File: src/reducer/authReducer.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+const authReducer = (state, dispatch) => {
+    console.log(state, dispatch);
```
Comment (parthpandyappp) [inline src/reducer/authReducer.js]: You can remove redundant logs from the final code.

**Comment by parthpandyappp** [comment]
I've put a comment above. Look into it. And yes, it looks good after you did destructuring.
Go ahead with the merge :+1:

**Review by parthpandyappp — APPROVED** [review]

File: src/reducer/filterReducer.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+export const filterReducer = (state, { ttype, payload }) => {
```
Comment (bishnoimukesh) [inline src/reducer/filterReducer.js]: ttype is used for?

**Review by bishnoimukesh — APPROVED** [review]
structured and clean code.
good to merge

File: src/reducer/authReducer.js (hunk 1 above)
Comment (yashpatel1797) [reply src/reducer/authReducer.js]: sure

File: src/reducer/filterReducer.js (hunk 2 above)
Comment (yashpatel1797) [reply src/reducer/filterReducer.js]: typo will change it.

**Pull request merged**